core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -99,13 +99,23 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
     // their requests and send them back to the pool for rescheduling (this may happen if "docker" operations
     // fail for example, or a container has aged and was destroying itself when a new request was assigned)
     case r: Run =>
-      // Check if the message is resent from the buffer. Only the first message on the buffer can be resent.
-      val isResentFromBuffer = runBuffer.nonEmpty && runBuffer.dequeueOption.exists(_._1.msg == r.msg)
+      // Checks if the current message is the first in the queue
+      val isFirstInQueue = runBuffer.dequeueOption.exists(_._1.msg == r.msg)
 
-      // Only process request, if there are no other requests waiting for free slots, or if the current request is the
-      // next request to process
-      // It is guaranteed, that only the first message on the buffer is resent.
-      if (runBuffer.isEmpty || isResentFromBuffer) {
+      if (!r.isFromQueue) {
+        // If the request is not from the queue, it will be added to the queue always.
+        runBuffer = runBuffer.enqueue(r)
+        runFirstMessageOfQueue()
+      } else if (r.isFromQueue && !isFirstInQueue) {
```

Comment:
If I'm not mistaken, all `Run` messages arriving at a proxy should have `fromQueue = true`. If such a `Run` message would exceed the proxies capacity, it's buffered instead of being run. If a run in the proxy fails, the proxy will destroy the container and send all buffered `Run` messages back to the pool. In that case, this code location may work on a `Run` message sent from a container and NOT being the first message enqueued. The comment states differently...